docs/secure-coding-guidelines.md

```diff
@@ -0,0 +1,96 @@
+# Secure Coding Guidelines
+
+## Purpose
+
+The secure coding guidelines are a reference to develop, deliver and maintain software in a secure manner.
+
+## Scope
+
+These guidelines apply to developers building client based applications at MetaMask.
+
+## References
+
+The guidelines in this policy were gathered from concepts on the [OWASP Top 10 Site](https://owasp.org/www-project-top-ten/) and modified to meet the client applications being developed at MetaMask.
+
+## Guidelines
+
+This document identifies six categories developers should reference when working on client based applications.
+
+1. User Input:
+
+   1. Processing User Inputs:
+      1. Always validate and sanitize user inputs
+      2. Preference should be given to libraries and frameworks that support input validation
```

Comment:
This may be something that should be moved down to dependencies. The general idea here is that is we are selecting libraries that process user data there should be a preference for the library/framework that performs validation over one that doesn't.